### PR TITLE
ci: Update Release Workflow

### PR DESCRIPTION
**Pull request by evansims**
### Changes

This PR updates the GH release workflow as part of our roll out process.

### References

N/A

### Testing

N/A

### Contributor Checklist

- [x] I agree to adhere to the [Auth0 General Contribution Guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md).
- [x] I agree to uphold the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### CODE REVIEW

File: .github/actions/get-version/action.yml
Comment (frederikprijck) [inline .github/actions/get-version/action.yml:4]: This won't work for repo's having multiple packages using different versions. Doesn't have to block merging this, but worth mentioning.

File: .github/workflows/release.yml
Comment (frederikprijck) [inline .github/workflows/release.yml:53]: Mostly informational, but if we have a PR with multiple commits, and we squash it in the release PR, would github.sha point to the new commit on the target branch (e.g. main / beta) ?

File: .github/actions/tag-create/action.yml
Comment (evansims) [reply .github/actions/tag-create/action.yml:29]: This will use the release merge commit from the branch the release PR is created from, so that should always be the case

File: .github/actions/get-version/action.yml
Comment (evansims) [reply .github/actions/get-version/action.yml:4]: 👍 For sure. Some repos will need some extra customizing for individual characteristics and circumstances like that.

File: .github/workflows/release.yml
Comment (evansims) [reply .github/workflows/release.yml:53]: Correct! This workflow is triggered after a release PR is merged, so the sha reflects the just-merged-commit to that target branch.

**Review by frederikprijck — APPROVED** [review]

**Pull request merged**
